pkg/validator/shoot_validator.go

```diff
@@ -0,0 +1,161 @@
+	if err := v.client.Get(ctx, kutil.Key(shoot.Spec.CloudProfileName), cloudProfile); err != nil {
+		return err
+	}
+
+	var (
+		allowedZones = getAllowedRegionZonesFromCloudProfile(shoot, cloudProfile)
+		workersPath  = fldPath.Child("workers")
+	)
+
+	controlPlaneConfigPath := fldPath.Child("controlPlaneConfig")
+	if shoot.Spec.Provider.ControlPlaneConfig == nil {
+		return field.Required(controlPlaneConfigPath, "ControlPlaneConfig must be set for GCP shoots")
+	}
+
+	controlPlaneConfig, err := decodeControlPlaneConfig(v.decoder, shoot.Spec.Provider.ControlPlaneConfig, controlPlaneConfigPath)
+	if err != nil {
+		return err
+	}
+
+	if errList := gcpvalidation.ValidateControlPlaneConfig(controlPlaneConfig, allowedZones); len(errList) != 0 {
+		return errList.ToAggregate()
+	}
+
+	if errList := gcpvalidation.ValidateWorkers(shoot.Spec.Provider.Workers, allowedZones, workersPath); len(errList) != 0 {
+		return errList.ToAggregate()
+	}
+
+	return nil
+}
+
+func (v *Shoot) validateShootUpdate(ctx context.Context, oldShoot, shoot *core.Shoot) error {
+	fldPath := field.NewPath("spec", "provider")
+
+	// InfrastructureConfig
+	if shoot.Spec.Provider.InfrastructureConfig == nil {
+		return field.Required(fldPath.Child("infrastructureConfig"), "InfrastructureConfig must be set for GCP shoots")
+	}
+
+	infraConfig, err := decodeInfrastructureConfig(v.decoder, shoot.Spec.Provider.InfrastructureConfig, fldPath)
+	if err != nil {
+		return err
+	}
+
+	if oldShoot.Spec.Provider.InfrastructureConfig == nil {
+		return field.InternalError(fldPath.Child("infrastructureConfig"), errors.New("InfrastructureConfig is not available on old shoot"))
+	}
+
+	oldInfraConfig, err := decodeInfrastructureConfig(v.decoder, oldShoot.Spec.Provider.InfrastructureConfig, fldPath)
+	if err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(oldInfraConfig, infraConfig) {
+		if errList := gcpvalidation.ValidateInfrastructureConfigUpdate(oldInfraConfig, infraConfig); len(errList) != 0 {
+			return errList.ToAggregate()
+		}
+	}
+
+	// ControlPlaneConfig
+	if shoot.Spec.Provider.ControlPlaneConfig == nil {
+		return field.Required(fldPath.Child("controlPlaneConfig"), "controlPlaneConfig must be set for GCP shoots")
+	}
+
+	if oldShoot.Spec.Provider.ControlPlaneConfig == nil {
+		return field.InternalError(fldPath.Child("controlPlaneConfig"), errors.New("controlPlaneConfig is not available on old shoot"))
+	}
+
+	controlPlaneConfig, err := decodeControlPlaneConfig(v.decoder, shoot.Spec.Provider.InfrastructureConfig, fldPath)
+	if err != nil {
+		return err
+	}
+
+	oldControlPlaneConfig, err := decodeControlPlaneConfig(v.decoder, oldShoot.Spec.Provider.ControlPlaneConfig, fldPath)
+	if err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(oldControlPlaneConfig, controlPlaneConfig) {
+		if errList := gcpvalidation.ValidateControlPlaneConfigUpdate(oldControlPlaneConfig, controlPlaneConfig); len(errList) != 0 {
+			return errList.ToAggregate()
+		}
+	}
+
```

Comment:
https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/validator/shoot_validator.go#L124-L134 should be added as well here

Reply:
Networks don't have Zones in GCP right?

Reply:
Hm, right. What about https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/validator/shoot_validator.go#L124-L134?

Reply:
+1
Applies to the `controlPlaneConfig` zone check.

Reply:
zone validation is done in `validateControlplaneConfig`

Reply:
I think we can resolve this comment since the zone in `ControlPlaneConfig` is immutable.